File: frontend/src/api/mission.ts
/** Mission commands: the two things an operator can tell the backend to do. */

import type { CommandResponse, MissionEvent, TelemetryFrame } from "../types";
import { get, post } from "./client";
import { endpoints } from "./endpoints";

export function authorize(): Promise<CommandResponse> {
  return post<CommandResponse>(endpoints.authorize());
}

export function reset(): Promise<CommandResponse> {
  return post<CommandResponse>(endpoints.reset());
}

/** Snapshot fallback. Live use goes through the telemetry socket. */
export function fetchTelemetry(): Promise<TelemetryFrame | null> {
  return get<TelemetryFrame | null>(endpoints.telemetry());
}

export function fetchEvents(): Promise<MissionEvent[]> {
  return get<MissionEvent[]>(endpoints.events());
}

File: backend/api/routers/mission.py
"""Operator mission commands.

Two endpoints, both intentionally thin: the interlocks live in the mission
state machine, not in the HTTP layer. A route that could authorize an
engagement on its own would be a second, weaker copy of that logic.
"""

from __future__ import annotations

from fastapi import APIRouter

from backend.api.deps import PipelineDep
from backend.schemas import CommandResponse

router = APIRouter(tags=["mission"])


@router.post("/api/authorize", response_model=CommandResponse)
async def authorize(pipeline: PipelineDep) -> CommandResponse:
    """Operator engagement authorization.

    The only action the operator must take during the demo. Rejected unless
    the mission has reached AWAITING_AUTHORIZATION.
    """
    ok, detail = pipeline.authorize()
    return CommandResponse(ok=ok, state=pipeline.mission.state, detail=detail)


@router.post("/api/reset", response_model=CommandResponse)
async def reset(pipeline: PipelineDep) -> CommandResponse:
    """Reset the mission so the full sequence can be demonstrated again."""
    pipeline.reset()
    return CommandResponse(
        ok=True,
        state=pipeline.mission.state,
        detail="Mission reset. Ready for a new run.",
    )
